packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -5647,6 +5647,44 @@ namespace Tpetra {
   }
 
   template <class LocalOrdinal, class GlobalOrdinal, class Node, const bool classic>
+  bool
+  CrsGraph<LocalOrdinal, GlobalOrdinal, Node, classic>::
+  doRowAndColumnMapsMatch ( ) const
+  {
+    using namespace Details;
+    const char tfecfFuncName[] = "doRowAndColumnMapsMatch: ";;
+
+    TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
+      (! isFillComplete (), std::runtime_error, "The graph must be fill complete.");
+
+    if (doRowAndColumnMapsMatch_ != MATCH_UNKNOWN)
+      return (doRowAndColumnMapsMatch_ == MATCH_TRUE);
+
+    auto rowLocalMap = rowMap_->getLocalMap();
+    auto colLocalMap = colMap_->getLocalMap();
+
+    const size_t numRows = rowLocalMap.getNodeNumElements();
+    const size_t numCols = colLocalMap.getNodeNumElements();
+
+    if (numCols < numRows) {
+      doRowAndColumnMapsMatch_ = MATCH_FALSE;
+      return false;
+    }
+
+    typedef Kokkos::RangePolicy<execution_space, LocalOrdinal> range_type;
+
+    size_t numDiff = 0;
+    Kokkos::parallel_reduce(range_type(0, numRows),
+      [this, rowLocalMap, colLocalMap](const LocalOrdinal i, size_t& diff) {
```

Comment:
Will capturing `this` (with all of its naughty non-CUDA-friendly fields) work with CUDA?  I don't think you even need to capture `this`; it's just a property of the two LocalMap objects.  See also the existing `isLocallyFitted` predicate (see above).